syntax = "proto3";

package PFDESolver;

option go_package = "/pfdesolver_protos";

message TClientConfig {
    optional uint64 SpaceCount = 1;                     // количество ячеек по x
    optional uint64 TimeCount = 2;                      // количество ячеек по t
    double LeftBound = 3;                               // левая граница отрезка сетки по x координате
    double RightBound = 4;                              // правая граница сетки по x координате
    double MaxTime = 5;                                 // граница отрезка времени по t координате
    double Alpha = 6;                                   // степень производой по x
    double Gamma = 7;                                   // степень производной по t
    double SpaceStep = 8;                               // шаг x по сетке
    double TimeStep = 9;                                // шаг t по сетке
    double Beta = 10;                                   // коэффициент доли лево/правосторонней производных [-1;+1]
    double AlphaLeft = 11;                              // коэффициент граничных условий третьего рода для x == L
    double BetaLeft = 12;                               // коэффициент граничных условий третьего рода для x == L
    double AlphaRight = 13;                             // коэффициент граничных условий третьего рода для x == R
    double BetaRight = 14;                              // коэффициент граничных условий третьего рода для x == R
    string DiffusionCoefficient = 15;                   // коэффициент диффузии
    string DemolitionCoefficient = 16;                  // коэффициент линейного сноса
    string ZeroTimeState = 17;                          // функция u(x, 0)
    string SourceFunction = 18;                         // функция f(x, t)
    string LeftBoundState = 19;                         // функция u(L, t)
    string RightBoundState = 20;                        // функция u(R, t)
    bool BordersAvailable = 21;                         // стоит ли учитывать граничные условия  
    optional uint64 StochasticIterationCount = 22;      // количество иттераций для стохастического метода
    optional string RealSolutionName = 23;              // названия истиного решения
    optional string RealSolution = 24;                  // функция u(x, t);

    optional string SolveMethods = 25;                  // методы для решения 
}


message TSolverConfig {
    uint64 SpaceCount = 1;      // количество ячеек по x
    uint64 TimeCount = 2;       // количество ячеек по t
    double LeftBound = 3;       // левая граница отрезка сетки по x координате
    double RightBound = 4;      // правая граница сетки по x координате
    double MaxTime = 5;         // граница отрезка времени по t координате
    double Alpha = 6;           // степень производой по x
    double Gamma = 7;           // степень производной по t
    double SpaceStep = 8;       // шаг x по сетке
    double TimeStep = 9;        // шаг t по сетке
    double Beta = 10;           // коэффициент доли лево/правосторонней производных [-1;+1]
    double AlphaLeft = 11;      // коэффициент граничных условий третьего рода для x == L
    double BetaLeft = 12;       // коэффициент граничных условий третьего рода для x == L
    double AlphaRight = 13;     // коэффициент граничных условий третьего рода для x == R
    double BetaRight = 14;      // коэффициент граничных условий третьего рода для x == R
    bool BordersAvailable = 15; // стоит ли учитывать граничные условия
}
